sw/device/silicon_creator/lib/shutdown.h

```diff
@@ -14,6 +14,22 @@
 extern "C" {
 #endif
 
+/**
+ * Evaluate an expression and call `shutdown_finalize` if the result is an
+ * error.
+ *
+ * The error will be passed as an argument to `shutdown_finalize`.
+ *
+ * @param expr_ An expression which results in an rom_error_t.
+ */
+#define SHUTDOWN_IF_ERROR(expr_) \
+  do {                           \
+    rom_error_t error = (expr_); \
```

Comment:
nit: `local_error_` instead of `error`?

Reply:
I ended up leaving this one because error is locally scoped so I don't think it is worth having a special name. Let me know if you disagree though, I don't feel strongly about it!